### PR TITLE
Made the default hitby() ignore observers

### LINKED ISSUE #16093: Haunting ghosts will act as projectiles when tossed out of a disposal chute and there is a windoor directly after the chute. Sanity issue.
#### Description of issue
Ghosts haunting a player going through a disposal chute will act as projectiles against a windoor should that windoor be on the tile directly after the outlet.
This appears to only affect windoors as well, as reinforced windows did not cause the ghosts to act as projectiles.
The only place where this is currently possible is security disposals segment because that one has a windoor directly after the outlet.

#### Difference between expected and actual behavior
Ghosts to have no effect on the windoor
Ghosts did have effect on the windoor, so much that just two ghosts haunting the person will destroy the windoor

#### Steps to reproduce 
Construct a disposal loop
On the tile directly after the outlet have a windoor placed
Haunt a body and admin throw it into the disposal loop
Watch it crash into the windoor as well as the ghosts

dreamseeker_2017-09-26_02-55-55

### PR DESCRIPTION
**Pull request by DamianX**
Fixes #16093

There may be another approach to fixing this I think this is alright.

### CODE REVIEW

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -173,8 +173,9 @@
 
 //When an object is thrown at the window
 /obj/machinery/door/window/hitby(AM as mob|obj)
-
-	..()
+	. = ..()
+	if(.)
+		return
```
Comment (N3X15) [inline code/game/machinery/doors/windowdoor.dm:178]:
```
if(..())
  return
```

File: code/game/atoms.dm [hunk 2]
```diff
@@ -563,8 +563,10 @@ its easier to just keep the beam vertical.
 	qdel(src)
 	return 1
 
-/atom/proc/hitby(atom/movable/AM as mob|obj)
-	return
+// Returns TRUE if it's been handled, children should return if parent has already handled
+/atom/proc/hitby(var/atom/movable/AM)
+	if(isobserver(AM))
+		. = TRUE
```
Comment (N3X15) [inline code/game/atoms.dm]: `return isobserver(AM)`

**Review by N3X15 — CHANGES_REQUESTED** [review]
You need to also test every hitby you changed, because a single screwup could result in invincibility issues.

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (DamianX) [reply code/game/machinery/doors/windowdoor.dm:178]: But that would return 0 to a child of a child. Is that alright?

Comment (N3X15) [reply code/game/machinery/doors/windowdoor.dm:178]: Correct, I am idort

**Comment by DamianX** [comment]
Works on my machine

**Review by N3X15 — APPROVED** [review]

**Pull request merged**